tiledb-sys/src/datatype.rs

```diff
@@ -207,6 +209,96 @@ impl Datatype {
             _ => None,
         }
     }
+
+    pub fn is_valid<T: 'static>(&self, _: &T) -> bool {
```

Comment:
"valid" meaning "`T` is a physical type which implements this logical type"?
I'd probably name it `is_physical_type` or at the least give put the "physical type" terminology in a doc comment

Reply:
I had some vague ideas on how this would work with slices that we'll need for strings and multi-value cells. I'll spend a bit this morning setting up an example to see if I can't decide whether we'll be able to use this same pattern for everything or if we'll have to get more complicated.

Which is to say, I'm not 100% that this will only work for physical types. Though I could see renaming to `is_compatible_type` or something.

Reply:
Experimental results show that we should be able to apply this same technique to non-physical types as well. According to the TypeId docs, it currently only works for types that are `'static` which I don't even know how it a type wouldn't be static yet. I'll go with `is_compatible_type` for now unless you throw something at me before I finish updating and merging.